### ISSUE TITLE
User Experience improvements

### ISSUE THREAD

**Issue by byronigoe** [kebernet/shortyz#116]
When I first got a smartphone I used Shortyz to do the NYTimes crossword. However, on many Thursday puzzles the answer for a single square is a word (aka Rebus: #35 feature). I tried many alternative apps and found Crosswords Plus, which not only supports Rebus (C in the attached picture), but also other usability improvements. Unfortunately, Crosswords Plus has not been updated in over 3 years, and now cannot download the NYTimes puzzles because of the backend changes. So, I'm back to Shortyz.

I believe Crosswords Plus "borrowed" from Shortyz when it was first developed, and it seems to be an abandoned project. So, I think it would be okay for us to duplicate some of its functionality.

I am a programmer with a couple published apps, and would be happy to help improve Shortyz. Before I just go making changes and submitting Pull Requests, I would like to have feedback from the community about these improvements.

![crosswordsplus](https://user-images.githubusercontent.com/10134042/30378477-51d50b86-9861-11e7-94e6-1963ebf4b95d.png)
A: Navigation to jump from clue to clue
B: See the clues for both directions simultaneously
C: Ability to enter multiple letters, numbers, and symbols into a single grid cell
D: Ability to Erase Wrong Letters in Word, and Erase All Wrong Answers.
E: The Clues view shows each clue followed by squares instead of one clue at a time:
![crosswordsplus2](https://user-images.githubusercontent.com/10134042/30379092-5ed00bb8-9863-11e7-91ff-9c5cd05aa3b3.png)

**Comment by byronigoe** [comment]
Two months later. Does nobody have an opinion on any of these items?

**Comment by markpilloff** [comment]
I switched from Shortyz to Crosswords Plus specifically because of item E
but grew to appreciate several of the other differences that you
highlighted. When NYT support went away I switched back to Shortyz. A while
ago I submitted a pull request to fix E (#106
<https://github.com/kebernet/shortyz/pull/106>) but it hasn't received any
attention.

Most recently I've switched to the official NYT app which has improved
quite a bit. I'd consider returning to Shortyz if you implemented the above
items, but I'm just a random user with no authority over the code, so take
that for what it's worth. :)

**Comment by gaul** [comment]
@byronigoe Opening individual issues are easier to address than omnibus ones like this.  Your suggestions seem valid but will take a long time to fix.

**Comment by ThatGitJer** [comment]
Yeah, there's not much activity here.  For me, I remained an observer until recently for 2 reasons: 
1) I couldn't be bothered to create a Github account
2) I was discouraged by the lack of activity here (which, of course, feeds the cycle)

I know Github is intended more for bug reporting than discussion, but it's disheartening to see so many posts unanswered.  Plus the list of reported bugs keeps growing, while the resolved bug count stays more-or-less unchanged.  I had bugs to report, but my feeling was "why bother?". 

Personally, I'd like to see the bugs fixed before any new features are added.  Unfortunately, Robert doesn't seem to have much time to spend on this project, so updates tend to be emergency fixes or new features (and I know from personal experience, it's far more enticing to add new code than to fix old code).

Anyway, as far as your suggestions go:
A: I am able to jump from clue to clue by swiping, or by tapping on the next clue
B: I can see clues for both directions simultaneously - maybe it's related to screen mode or ratio?  (I use a 7" 16:10 tablet in portrait mode)
C: I'd like multi-letter 'rebus' squares; I haven't ever needed numbers or symbols
D: might be useful; wouldn't hurt to add this
E: I never use the Clues view, so no comment (again, maybe related to how I've got the screen laid out)

Overall, my gut reaction was: "Ack! Change bad!"  (And looking back on Shortyz release notes, this seems 
to be a common sentiment).  For me, it's partly about getting old(er) & lazy, as well as seeing too much software made worse [IMHO] by changes (e.g. Windows, Firefox, certain areas of Android).  Also, Shortyz has a history of changes and new features being accompanied by bugs (both new & resurfaced old ones).

But, if you could make these changes unobtrusively (i.e. without breaking existing code, or forcing people to change the way they currently use Shortyz), I guess we'd be no worse off.  And if you could fix a few bugs while you're digging through the code, you might even make more friends than enemies. ;-)  
[of course, you're more likely to hear from the latter]

My top 5 bugs that need fixing: (if anyone cares)
- puzzles getting stuck "over-scrolled" after leaving a clue that requires multiple lines
- accented characters in Newsday clues not showing  [wrong charset, I assume]
- erratic & unpredictable move-to-next-word behavior (after typing the last letter of current word)
- some colors not right since Night Mode was added (maybe Marshmallow-specific)
- Chronicle of Higher Education no longer downloading?  [urls haven't changed, maybe HTTP Referer issue]

A couple more thoughts: 
- have you tried out alphaCross? - https://play.google.com/store/apps/details?id=org.akop.crosswords
I actually haven't tried it myself (see above for my attitude towards change).  But it looked promising enough that I added it to my App Watcher list, and it gets updated quite often.  And it's got some of the features you suggested.
- if you have grand ambitions, you might want to consider forking into a separate project

Well, you wanted thoughts, so those are mine (some are even on-topic!).

TL;DR: Don't break anything, or force people to change their ways.  Otherwise, yeah, go for it.
Try to fix some existing bugs.  Oh, and email Robert first, since he'll be the one getting any hate-mail. ;-)

Jeremy
